src/row.rs

```diff
@@ -0,0 +1,113 @@
+/// A macro for creating a [`Row`] using vec! syntax.
+///
+/// `row!` is similar to the [`vec!`] macro, but it returns a [`Row`] instead of a `Vec`.
+///
+/// # Examples
+///
+/// * Create a [`Row`] containing a vector of [`Cell`]s:
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::row;
+///
+/// let row = row!["hello", "world"];
+/// let row = row!["hello".red(), "world".red().bold()];
+/// ```
+///
+/// * Create a [`Row`] from a given [`Cell`] repeated some amount of times:
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::row;
+///
+/// let row = row!["hello"; 2];
+/// ```
+///
+/// * Use [`text!`], [`line!`] or [`span!`] macro inside [`row!`] macro.
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::{row, line, text, span};
+///
+/// let row = row![
+///     line!["hello", "world"], span!(Modifier::BOLD; "goodbye {}", "world"),
+///     text!["hello", "world"],
+/// ];
+/// ```
+///
+/// [`Row`]: crate::widgets::Row
+/// [`Cell`]: crate::widgets::Cell
+#[macro_export]
+macro_rules! row {
```

Comment:
Q: Do you think this would be worth naming table_row? My thinking being this name is fairly short and macros often end up in scope all over the place. (I think the answer is likely no, but it's worth asking).

Reply:
I don't think it's worth it because the advantage here is to make it shorter and easier to write imo. And `row!` implies a row in a table?

Reply:
I don't disagree, but the context for my thinking is that in Ratatui, we re-export `ratatui::widgets::table::{Table, Row, Cell}` as `ratatui::widgets::{Table, Row, Cell}`, and I think these probably should have been prefixed if we do that `ratatui::widgets::table::{Table, TableRow, TableCell}`, or just not exported (and used as `table::Row, table::Cell` etc. It's not a big deal though and the ship has probably sailed on that one.